From 7ada4b7e39819f2cc765b27f3fc1ad38ece3278f Mon Sep 17 00:00:00 2001
From: Tomer Maimon <tmaimon77@gmail.com>
Date: Mon, 1 Jul 2024 10:10:46 +0300
Subject: [PATCH] arm64: dts: modify clock property in modules node

Modify clock property handler in UART, CPU, PECI modules to reset
controller.

This change was pushed upstream and under reviewing:
https://lore.kernel.org/all/20240701071048.751863-5-tmaimon77@gmail.com/

Signed-off-by: Tomer Maimon <tmaimon77@gmail.com>
---
 .../boot/dts/nuvoton/nuvoton-common-npcm8xx.dtsi | 16 ++++++++--------
 arch/arm64/boot/dts/nuvoton/nuvoton-npcm845.dtsi |  8 ++++----
 2 files changed, 12 insertions(+), 12 deletions(-)

diff --git a/arch/arm64/boot/dts/nuvoton/nuvoton-common-npcm8xx.dtsi b/arch/arm64/boot/dts/nuvoton/nuvoton-common-npcm8xx.dtsi
index 41d345448430..92e3b0fe746f 100644
--- a/arch/arm64/boot/dts/nuvoton/nuvoton-common-npcm8xx.dtsi
+++ b/arch/arm64/boot/dts/nuvoton/nuvoton-common-npcm8xx.dtsi
@@ -73,7 +73,7 @@ peci: peci-controller@100000 {
 				compatible = "nuvoton,npcm845-peci";
 				reg = <0x100000 0x1000>;
 				interrupts = <GIC_SPI 6 IRQ_TYPE_LEVEL_HIGH>;
-				clocks = <&clk NPCM8XX_CLK_APB3>;
+				clocks = <&rstc NPCM8XX_CLK_APB3>;
 				cmd-timeout-ms = <1000>;
 				status = "disabled";
 			};
@@ -89,7 +89,7 @@ timer0: timer@8000 {
 			serial0: serial@0 {
 				compatible = "nuvoton,npcm845-uart", "nuvoton,npcm750-uart";
 				reg = <0x0 0x1000>;
-				clocks = <&clk NPCM8XX_CLK_UART>;
+				clocks = <&rstc NPCM8XX_CLK_UART>;
 				interrupts = <GIC_SPI 192 IRQ_TYPE_LEVEL_HIGH>;
 				reg-shift = <2>;
 				status = "disabled";
@@ -98,7 +98,7 @@ serial0: serial@0 {
 			serial1: serial@1000 {
 				compatible = "nuvoton,npcm845-uart", "nuvoton,npcm750-uart";
 				reg = <0x1000 0x1000>;
-				clocks = <&clk NPCM8XX_CLK_UART>;
+				clocks = <&rstc NPCM8XX_CLK_UART>;
 				interrupts = <GIC_SPI 193 IRQ_TYPE_LEVEL_HIGH>;
 				reg-shift = <2>;
 				status = "disabled";
@@ -107,7 +107,7 @@ serial1: serial@1000 {
 			serial2: serial@2000 {
 				compatible = "nuvoton,npcm845-uart", "nuvoton,npcm750-uart";
 				reg = <0x2000 0x1000>;
-				clocks = <&clk NPCM8XX_CLK_UART>;
+				clocks = <&rstc NPCM8XX_CLK_UART>;
 				interrupts = <GIC_SPI 194 IRQ_TYPE_LEVEL_HIGH>;
 				reg-shift = <2>;
 				status = "disabled";
@@ -116,7 +116,7 @@ serial2: serial@2000 {
 			serial3: serial@3000 {
 				compatible = "nuvoton,npcm845-uart", "nuvoton,npcm750-uart";
 				reg = <0x3000 0x1000>;
-				clocks = <&clk NPCM8XX_CLK_UART>;
+				clocks = <&rstc NPCM8XX_CLK_UART>;
 				interrupts = <GIC_SPI 195 IRQ_TYPE_LEVEL_HIGH>;
 				reg-shift = <2>;
 				status = "disabled";
@@ -125,7 +125,7 @@ serial3: serial@3000 {
 			serial4: serial@4000 {
 				compatible = "nuvoton,npcm845-uart", "nuvoton,npcm750-uart";
 				reg = <0x4000 0x1000>;
-				clocks = <&clk NPCM8XX_CLK_UART>;
+				clocks = <&rstc NPCM8XX_CLK_UART>;
 				interrupts = <GIC_SPI 196 IRQ_TYPE_LEVEL_HIGH>;
 				reg-shift = <2>;
 				status = "disabled";
@@ -134,7 +134,7 @@ serial4: serial@4000 {
 			serial5: serial@5000 {
 				compatible = "nuvoton,npcm845-uart", "nuvoton,npcm750-uart";
 				reg = <0x5000 0x1000>;
-				clocks = <&clk NPCM8XX_CLK_UART>;
+				clocks = <&rstc NPCM8XX_CLK_UART>;
 				interrupts = <GIC_SPI 197 IRQ_TYPE_LEVEL_HIGH>;
 				reg-shift = <2>;
 				status = "disabled";
@@ -143,7 +143,7 @@ serial5: serial@5000 {
 			serial6: serial@6000 {
 				compatible = "nuvoton,npcm845-uart", "nuvoton,npcm750-uart";
 				reg = <0x6000 0x1000>;
-				clocks = <&clk NPCM8XX_CLK_UART>;
+				clocks = <&rstc NPCM8XX_CLK_UART2>;
 				interrupts = <GIC_SPI 198 IRQ_TYPE_LEVEL_HIGH>;
 				reg-shift = <2>;
 				status = "disabled";
diff --git a/arch/arm64/boot/dts/nuvoton/nuvoton-npcm845.dtsi b/arch/arm64/boot/dts/nuvoton/nuvoton-npcm845.dtsi
index 383938dcd3ce..3cbcea65eba2 100644
--- a/arch/arm64/boot/dts/nuvoton/nuvoton-npcm845.dtsi
+++ b/arch/arm64/boot/dts/nuvoton/nuvoton-npcm845.dtsi
@@ -14,7 +14,7 @@ cpus {
 		cpu0: cpu@0 {
 			device_type = "cpu";
 			compatible = "arm,cortex-a35";
-			clocks = <&clk NPCM8XX_CLK_CPU>;
+			clocks = <&rstc NPCM8XX_CLK_CPU>;
 			reg = <0x0 0x0>;
 			next-level-cache = <&l2>;
 			enable-method = "psci";
@@ -23,7 +23,7 @@ cpu0: cpu@0 {
 		cpu1: cpu@1 {
 			device_type = "cpu";
 			compatible = "arm,cortex-a35";
-			clocks = <&clk NPCM8XX_CLK_CPU>;
+			clocks = <&rstc NPCM8XX_CLK_CPU>;
 			reg = <0x0 0x1>;
 			next-level-cache = <&l2>;
 			enable-method = "psci";
@@ -32,7 +32,7 @@ cpu1: cpu@1 {
 		cpu2: cpu@2 {
 			device_type = "cpu";
 			compatible = "arm,cortex-a35";
-			clocks = <&clk NPCM8XX_CLK_CPU>;
+			clocks = <&rstc NPCM8XX_CLK_CPU>;
 			reg = <0x0 0x2>;
 			next-level-cache = <&l2>;
 			enable-method = "psci";
@@ -41,7 +41,7 @@ cpu2: cpu@2 {
 		cpu3: cpu@3 {
 			device_type = "cpu";
 			compatible = "arm,cortex-a35";
-			clocks = <&clk NPCM8XX_CLK_CPU>;
+			clocks = <&rstc NPCM8XX_CLK_CPU>;
 			reg = <0x0 0x3>;
 			next-level-cache = <&l2>;
 			enable-method = "psci";
-- 
2.34.1

